saleor/graphql/order/bulk_mutations/order_bulk_create.py

```diff
@@ -930,55 +1232,85 @@ def create_single_note(
         if any([note_input.get(key) for key in user_key_map.keys()]):
             user = cls.get_instance_with_errors(
                 input=note_input,
-                errors=errors,
+                errors=order_data.errors,
                 model=User,
                 key_map=user_key_map,
                 object_storage=object_storage,
+                path=f"notes.{index}",
             )
 
         if note_input.get("app_id"):
             app = cls.get_instance_with_errors(
                 input=note_input,
-                errors=errors,
+                errors=order_data.errors,
                 model=App,
                 key_map={"app_id": "id"},
                 object_storage=object_storage,
+                path=f"notes.{index}",
             )
 
         if user and app:
             user, app = None, None
-            errors.append(
+            order_data.errors.append(
                 OrderBulkError(
                     message="Note input contains both user and app identifier.",
                     code=OrderBulkCreateErrorCode.TOO_MANY_IDENTIFIERS,
-                    field="notes",
+                    path=f"notes.{index}",
                 )
             )
 
         event = OrderEvent(
             date=date,
             type=OrderEvents.NOTE_ADDED,
-            order=order,
+            order=order_data.order,
             parameters={"message": note_input["message"]},
             user=user,
             app=app,
         )
 
         return event
 
+    @classmethod
+    def create_single_discount(
+        cls,
+        discount_input: Dict[str, Any],
+        order_data: OrderBulkCreateData,
+        order_amounts: OrderAmounts,
+        currency: str,
+        index: int,
+    ) -> OrderDiscount:
+        max_total = Money(order_amounts.undiscounted_total_gross, currency)
+        try:
+            OrderDiscountCommon.validate_order_discount_input(max_total, discount_input)
+        except ValidationError as err:
+            order_data.errors.append(
+                OrderBulkError(
+                    message=err.messages[0],
+                    path=f"discounts.{index}",
+                    code=OrderBulkCreateErrorCode.INVALID,
+                )
+            )
+
+        return OrderDiscount(
+            order=order_data.order,
+            value_type=discount_input["value_type"],
+            value=discount_input["value"],
+            reason=discount_input.get("reason"),
+        )
```

Comment:
Shouldn't we provide also `amount`?

Reply:
According to RFC discount input is type of `OrderDiscountCommonInput` and there is no info about amount. In general, for `OrderBulkCreate` mutation, `lineTotalPrice` and `undiscountedLineTotalPrice` are the primary sources of truth about the order pricing.